### ISSUE TITLE
A11y warnings do not account for destructured attributes.

### ISSUE THREAD

**Issue by iolyd** [sveltejs/language-tools#2135]
### Describe the bug

I haven't tested with different cases yet, but assigning an anchor tag's href using a destructured object does not appear to mitigate missing attribute warnings.

### Reproduction

For illustration:

```svelte
<script lang="ts">
  const attributes = {
    href: '/about'
  }
</script>

<nav>
  <a {...attributes}>About</a> <!-- A11y: <a> element should have an href attributesvelte(a11y-missing-attribute) -->
</nav>
```

Cases like the above will quickly be met when using an approach similar to melt-ui's builders to manage an element's attributes. For example, in my app I have a series of small i18n helpers that localize anchor `hrefs` (prepending the locale param) and specify their corresponding `hreflang` attribute:

```svelte
<nav>
  <a {...$i18nlink('/about')}>{t('about')}</a> <!-- A11y: <a> element should have an href attributesvelte(a11y-missing-attribute) -->
  <a href={$i18nhref('/projects')} hreflang={$page.data.locale}>{t('projects')}</a>
</nav>
```

where `i18nlink` is a derived store returning an object `{href: string, hreflang: string}`.

### Expected behaviour

If destructured object used for element attributes contains expected attribute keys (for example `href` on an anchor tag), omit warning.

### System Info

- OS: MacOS
- IDE: VSCode

### Which package is the issue about?

Svelte for VS Code extension
